vortex-array/src/array/bool/compute/fill.rs

```diff
@@ -8,11 +8,18 @@ use crate::{Array, ArrayDType, IntoArray};
 
 impl FillForwardFn for BoolArray {
     fn fill_forward(&self) -> VortexResult<Array> {
-        if self.dtype().nullability() == Nullability::NonNullable {
+        let validity = self.logical_validity();
+        if self.dtype().nullability() == Nullability::NonNullable
+            || validity.all_valid()
+            || validity.all_invalid()
```

Comment:
I think we should figure out what's the contract of fillforward. Just to note that there's a subtle change where before AllValid would end up as NonNullable array while now will remain Nullable

Reply:
yeah, I opened #591 as a FLUP because I ran into inconsistent behavior on various implementations of `slice` as well. We should audit/decide/document/enforce consistently

Reply:
For now, I'm going to go ahead and change fill_forward to be `Validity::NonNullable` iff dtype is NonNullable; otherwise Validity will be guaranteed to be `Validity::AllValid`